ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/AlterTableHandler.java

```diff
@@ -92,21 +93,49 @@ private Scenario scenarioType(org.apache.hadoop.hive.metastore.api.Table before,
     }
   }
 
-  // return true, if event needs to be dumped, else return false.
-  private boolean handleForTableLevelReplication(Context withinContext) {
-    String oldName = before.getTableName();
-    String newName = after.getTableName();
+  // Return true, if event needs to be dumped, else return false.
+  private boolean handleRenameForReplacePolicy(Context withinContext, String oldName, String newName) {
+    // If the table is renamed after being added to the list of tables to be bootstrapped, then remove it from the
```

Comment:
Just add an assert that withinContext.oldReplScope != null.

Reply:
with the addition if withinContext.oldReplScope != null ..before calling ..the assert is not required